connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/RestClient.java

```diff
@@ -67,13 +68,13 @@ HttpClient httpClient(SslContextFactory sslContextFactory) {
     /**
      * Sends HTTP request to remote REST server
      *
-     * @param url             HTTP connection will be established with this url.
-     * @param method          HTTP method ("GET", "POST", "PUT", etc.)
+     * @param url             HTTP connection will be established with this url, non-null.
+     * @param method          HTTP method ("GET", "POST", "PUT", etc.), non-null
      * @param headers         HTTP headers from REST endpoint
      * @param requestBodyData Object to serialize as JSON and send in the request body.
-     * @param responseFormat  Expected format of the response to the HTTP request.
+     * @param responseFormat  Expected format of the response to the HTTP request, non-null.
      * @param <T>             The type of the deserialized response to the HTTP request.
-     * @return The deserialized response to the HTTP request, or null if no data is expected.
+     * @return The deserialized response to the HTTP request, containing null if no data is expected or returned.
```

Comment:
"no data is expected" sounds like an odd description for a return value, could we just remove that bit?